### DISCUSSION TITLE
Reference Point Lalo to XY conversion

### DISCUSSION THREAD

**Discussion by zamuzakki** [insarlab/MintPy#1203] [Q&A]
Good day!

I used Miaplpy and Mintpy for InSAR processing. In the Miaplpy settings, I set `mintpy.reference.lalo = -25.45988630463442, 27.132471386336594`
Mintpy then converts it to (273, 892), as indicated in these logs

> input reference point in lat/lon: (-25.46065748228001, 27.13239301372493)
input reference point in y/x: (273, 892)

When I checked the KML output, there is discrepancy between the provided lat long reference point and the marking

Actual lat long:
![Screenshot_20240523_101403](https://github.com/insarlab/MintPy/assets/7352963/61daf6c6-73d4-4ad7-badd-90cd7afb7080)

Marked reference point
![Screenshot_20240523_101506](https://github.com/insarlab/MintPy/assets/7352963/21a38485-85b0-4fa9-b3df-a38617311ac0)

Is this discrepancy something to be expected?

**Comment by welcome** [comment]
👋  Thanks for opening your first issue here! Please filled out the template with as much details as possible. We appreciate that you took the time to contribute!
Make sure you read our [contributing guidelines](https://github.com/insarlab/MintPy/blob/master/docs/CONTRIBUTING.md).

**Comment by yunjunz** [comment]
Hi @zamuzakki, yes, this is expected. mintpy convert the lat/lon to y/x (row/col in radar coordinates) using the following approach (https://nbviewer.org/github/insarlab/MintPy-tutorial/blob/main/applications/coord_conversion.ipynb), and then use the lat/lon at the center of the determined pixel, to ensure the actually used y/x and lat/lon of the reference point is exactly consistent.

This brings two potential discrepancies from the user's input lat/lon:
1. the user specified lat/lon could not be at the pixel center, thus, different from the resulting lat/lon at the sub-pixel level.
2. the lat/lon to y/x conversion could be inaccurate at the level of a few pixels.

The third potential source of the discrepancy you saw here is the absolute geolocation accuracy of the SAR data itself. Without special corrections, it could be meters due to DEM error and tens of cm due to other physical sources ([Gisinger et al., 2021](https://ieeexplore.ieee.org/document/9115211); [Yunjun et al., 2022](https://ieeexplore.ieee.org/document/9759304)).
